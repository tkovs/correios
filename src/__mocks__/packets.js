const statuses = [
  {
    status: 'Objeto entregue ao destinat치rio',
    datetime: {
      date: {
        day: 4,
        month: 2,
        year: 2020,
      },
      time: {
        hour: 16,
        minute: 37,
      },
    },
    locale: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto saiu para entrega ao destinat치rio',
    datetime: {
      date: {
        day: 4,
        month: 2,
        year: 2020,
      },
      time: {
        hour: 11,
        minute: 38,
      },
    },
    locale: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto encaminhado ',
    datetime: {
      date: {
        day: 2,
        month: 2,
        year: 2020,
      },
      time: {
        hour: 9,
        minute: 51,
      },
    },
    from: {
      place: 'CTE BENFICA',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
    to: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto encaminhado ',
    datetime: {
      date: {
        day: 1,
        month: 2,
        year: 2020,
      },
      time: {
        hour: 10,
        minute: 57,
      },
    },
    from: {
      place: 'CTE BENFICA',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
    to: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto encaminhado ',
    datetime: {
      date: {
        day: 31,
        month: 1,
        year: 2020,
      },
      time: {
        hour: 11,
        minute: 1,
      },
    },
    from: {
      place: 'CTCE CURITIBA',
      city: 'Curitiba',
      state: 'PR',
    },
    to: {
      place: 'CTE BENFICA',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto encaminhado ',
    datetime: {
      date: {
        day: 30,
        month: 1,
        year: 2020,
      },
      time: {
        hour: 8,
        minute: 52,
      },
    },
    from: {
      place: 'AC ASSIS CHATEAUBRIAND',
      city: 'Assis Chateaubriand',
      state: 'PR',
    },
    to: {
      place: 'CTCE CURITIBA',
      city: 'Curitiba',
      state: 'PR',
    },
  },
  {
    status: 'Objeto postado ap칩s o hor치rio limite da unidade',
    datetime: {
      date: {
        day: 29,
        month: 1,
        year: 2020,
      },
      time: {
        hour: 16,
        minute: 46,
      },
    },
    locale: {
      place: 'AC ASSIS CHATEAUBRIAND',
      city: 'Assis Chateaubriand',
      state: 'PR',
    },
  },
]

export default [
  {
    title: 'iPhone 7',
    status: 'delivered',
    code: 'PW086958109BR',
    mode: 'SEDEX',
    statuses,
  },
]
