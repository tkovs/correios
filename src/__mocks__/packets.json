[
  {
    "title": "iPad",
    "status": "delivered",
    "code": "PW086958100BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "Apple Pencil",
    "status": "delivered",
    "code": "PW086958101BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  },
  {
    "title": "Macbook Pro",
    "status": "delivered",
    "code": "PW086958102BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "Magic Mouse",
    "status": "delivered",
    "code": "PW086958103BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  },
  {
    "title": "iMac",
    "status": "delivered",
    "code": "PW086958104BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "iPhone",
    "status": "delivered",
    "code": "PW086958105BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  },
  {
    "title": "iPod",
    "status": "delivered",
    "code": "PW086958106BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "Applewatch",
    "status": "delivered",
    "code": "PW086958107BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  },
  {
    "title": "Dell Vostro 5470",
    "status": "delivered",
    "code": "PW086958108BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "Peixinho da horta",
    "status": "delivered",
    "code": "PW086958109BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  },
  {
    "title": "Camiseta C&A",
    "status": "delivered",
    "code": "PW086958110BR",
    "mode": "PAC",
    "createdAt": "05/02/2020",
    "updatedAt": "09/02/2020",
    "deliveredAt": "09/02/2020"
  },
  {
    "title": "Bola de beisebol",
    "status": "delivered",
    "code": "PW086958111BR",
    "mode": "SEDEX",
    "createdAt": "07/02/2020",
    "updatedAt": "10/02/2020",
    "deliveredAt": "10/02/2020"
  }
]
