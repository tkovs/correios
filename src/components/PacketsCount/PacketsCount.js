import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'
import { Caption } from 'react-native-paper'
import { colors } from '../../styles/theme'

function PacketsCount({ quantity, testID }) {
  if (quantity === 0) {
    return null
  }

  const phrase = quantity === 1 ? '1 encomenda' : `${quantity} encomendas`

  return (
    <View style={styles.container}>
      <Caption style={styles.text} testID={testID}>
        {phrase}
      </Caption>
    </View>
  )
}

PacketsCount.defaultProps = {
  quantity: 0,
  testID: null,
}

PacketsCount.propTypes = {
  quantity: PropTypes.number,
  testID: PropTypes.string,
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.sandstorm,
    alignItems: 'center',
    borderBottomColor: colors.yellow,
    borderBottomWidth: 1,
  },
  text: {
    color: colors.blue,
  },
})

export default PacketsCount
