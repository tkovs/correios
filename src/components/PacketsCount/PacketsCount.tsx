import React, { FC } from 'react'
import { View, StyleSheet } from 'react-native'
import { Caption } from 'react-native-paper'
import { colors } from '../../styles/theme'

interface PacketsCountProps {
  quantity: number
  testID?: string
}

type Props = PacketsCountProps

const PacketsCount: FC<Props> = ({ quantity, testID }: Props) => {
  if (quantity === 0) {
    return null
  }

  const phrase = quantity === 1 ? '1 encomenda' : `${quantity} encomendas`

  return (
    <View style={styles.container}>
      <Caption style={styles.text} testID={testID}>
        {phrase}
      </Caption>
    </View>
  )
}

PacketsCount.defaultProps = {
  quantity: 0,
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.sandstorm,
    alignItems: 'center',
    borderBottomColor: colors.yellow,
    borderBottomWidth: 1,
  },
  text: {
    color: colors.blue,
  },
})

export default PacketsCount
