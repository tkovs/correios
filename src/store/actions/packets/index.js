import * as types from './types'

export const addPacket = (title, code) => ({
  type: types.ADD_PACKET_PENDING,
  payload: {
    title,
    code,
  },
})

export const addPacketSuccess = packet => ({
  type: types.ADD_PACKET_SUCCESS,
  payload: {
    packet,
  },
})

export const addPacketFailure = error => ({
  type: types.ADD_PACKET_FAILURE,
  payload: {
    error,
  },
})

export const clearError = () => ({
  type: types.CLEAR_ERROR,
})
