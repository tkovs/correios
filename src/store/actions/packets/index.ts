import {
  ADD_PACKET_PENDING,
  ADD_PACKET_SUCCESS,
  ADD_PACKET_FAILURE,
  UPDATE_PACKETS,
  UPDATE_PACKET_PENDING,
  UPDATE_PACKET_SUCCESS,
  UPDATE_PACKET_FAILURE,
  REMOVE_PACKET,
  ARCHIVE_PACKET,
  UPDATE_LAST_VIEW,
  PacketsActionTypes,
} from './types'

export const addPacket = (title: string, code: string): PacketsActionTypes => ({
  type: ADD_PACKET_PENDING,
  payload: {
    title,
    code,
  },
})

export const addPacketSuccess = (packet: Packet): PacketsActionTypes => ({
  type: ADD_PACKET_SUCCESS,
  payload: {
    packet,
  },
})

export const addPacketFailure = (
  error: string,
  code: string
): PacketsActionTypes => ({
  type: ADD_PACKET_FAILURE,
  payload: {
    error,
    code,
  },
})

export const updatePackets = (): PacketsActionTypes => ({
  type: UPDATE_PACKETS,
})

export const updatePacket = (packet: Packet): PacketsActionTypes => ({
  type: UPDATE_PACKET_PENDING,
  payload: {
    packet,
  },
})

export const updatePacketSuccess = (packet: Packet): PacketsActionTypes => ({
  type: UPDATE_PACKET_SUCCESS,
  payload: {
    packet,
  },
})

export const updatePacketFailure = (
  error: string,
  code: string
): PacketsActionTypes => ({
  type: UPDATE_PACKET_FAILURE,
  payload: {
    error,
    code,
  },
})

export const removePacket = (code: string): PacketsActionTypes => ({
  type: REMOVE_PACKET,
  payload: {
    code,
  },
})

export const archivePacket = (code: string): PacketsActionTypes => ({
  type: ARCHIVE_PACKET,
  payload: {
    code,
  },
})

export const updateLastView = (
  date: Date,
  code: string
): PacketsActionTypes => ({
  type: UPDATE_LAST_VIEW,
  payload: {
    date,
    code,
  },
})
