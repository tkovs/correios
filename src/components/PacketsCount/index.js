import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import PacketsCount from './PacketsCount'
import { packetsListWithoutArchivedSelector } from '../../store/selectors/packets'

function PacketsCountContainer({ packets, filter, testID }) {
  const quantity = filter
    ? packets.filter(packet => packet.status === filter).length
    : packets.length

  return <PacketsCount quantity={quantity} testID={testID} />
}

PacketsCountContainer.defaultProps = {
  filter: null,
  testID: null,
}

PacketsCountContainer.propTypes = {
  packets: PropTypes.arrayOf(PropTypes.object).isRequired,
  filter: PropTypes.string,
  testID: PropTypes.string,
}

const mapStateToProps = state => ({
  packets: packetsListWithoutArchivedSelector(state),
})

export default connect(mapStateToProps)(PacketsCountContainer)
