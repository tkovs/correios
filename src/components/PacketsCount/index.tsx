import React, { FC } from 'react'
import { connect, MapStateToProps } from 'react-redux'

import PacketsCount from './PacketsCount'
import { packetsListWithoutArchivedSelector } from '../../store/selectors/packets'

interface StateProps {
  packets: Packet[]
}

interface OwnProps {
  filter?: string
  testID?: string
}

type Props = StateProps & OwnProps

const PacketsCountContainer: FC<Props> = ({
  packets,
  filter,
  testID,
}: Props) => {
  const quantity = filter
    ? packets.filter((packet: Packet) => packet.status === filter).length
    : packets.length

  return <PacketsCount quantity={quantity} testID={testID} />
}

const mapStateToProps: MapStateToProps<StateProps, OwnProps, State> = (
  state: State
) => ({
  packets: packetsListWithoutArchivedSelector(state),
})

export default connect(mapStateToProps)(PacketsCountContainer)
