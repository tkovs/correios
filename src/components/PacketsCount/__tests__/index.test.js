import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'
import { createMockStore } from 'redux-logic-test'

import PacketsCount from '..'
import rootReducer from '../../../store/reducers'
import rootLogic from '../../../store/logic'
import {
  getComponentWithPaper,
  getComponentWithRedux,
} from '../../../utils/jest'
import { addPacketSuccess } from '../../../store/actions/packets'
import { statuses } from '../../../__mocks__/packets'

let store = null

describe('PacketsCount', () => {
  beforeEach(() => {
    store = createMockStore({ logic: rootLogic, reducer: rootReducer })
    store.resetActions()
  })

  const mockTestID = 'mockTestID'

  describe('with quantity 0', () => {
    it('renders correctly', () => {
      const { baseElement } = render(
        getComponentWithPaper(
          getComponentWithRedux(store, <PacketsCount tetsID={mockTestID} />)
        )
      )
      expect(store.getState().packets.list.length).toEqual(0)
      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('with quantity 1', () => {
    it('renders correctly', async () => {
      store.dispatch(addPacketSuccess({ code: 'mockCode', statuses }))

      await store.whenComplete(() => {
        const { baseElement } = render(
          getComponentWithPaper(
            getComponentWithRedux(store, <PacketsCount tetsID={mockTestID} />)
          )
        )
        expect(store.getState().packets.list.length).toEqual(1)
        expect(baseElement).toMatchSnapshot()
      })
    })
  })

  describe('with more quantity than 1', () => {
    it('renders correctly', async () => {
      store.dispatch(addPacketSuccess({ code: 'mockCode1', statuses }))
      store.dispatch(addPacketSuccess({ code: 'mockCode2', statuses }))

      await store.whenComplete(() => {
        const { baseElement } = render(
          getComponentWithPaper(
            getComponentWithRedux(store, <PacketsCount tetsID={mockTestID} />)
          )
        )
        expect(store.getState().packets.list.length).toEqual(2)
        expect(baseElement).toMatchSnapshot()
      })
    })
  })
})
