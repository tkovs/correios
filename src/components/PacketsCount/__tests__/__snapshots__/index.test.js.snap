// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacketsCount with more quantity than 1 renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEC643",
            "borderBottomColor": "#F9D900",
            "borderBottomWidth": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {
                    "color": "#01417F",
                  },
                ],
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {
                    "color": "#01417F",
                  },
                ],
              ],
            ]
          }
        >
          2 encomendas
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PacketsCount with quantity 0 renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`PacketsCount with quantity 1 renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEC643",
            "borderBottomColor": "#F9D900",
            "borderBottomWidth": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {
                    "color": "#01417F",
                  },
                ],
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {
                    "color": "#01417F",
                  },
                ],
              ],
            ]
          }
        >
          1 encomenda
        </Text>
      </View>
    </View>
  </View>
</View>
`;
